types/blob/blob.go

```diff
@@ -0,0 +1,150 @@
+package blob
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/namespace"
+	rpcproto "github.com/rollkit/celestia-openrpc/proto/blob/rollkit"
+)
+
+const (
+	// NMTIgnoreMaxNamespace is currently used value for IgnoreMaxNamespace option in NMT.
+	// IgnoreMaxNamespace defines whether the largest possible namespace.ID MAX_NID should be 'ignored'.
+	// If set to true, this allows for shorter proofs in particular use-cases.
+	NMTIgnoreMaxNamespace = true
+)
+
+var (
+	ErrBlobNotFound = errors.New("blob: not found")
+	ErrInvalidProof = errors.New("blob: invalid proof")
+)
+
+// Proof is a collection of nmt.Proofs that verifies the inclusion of the data.
+type Proof []*nmt.Proof
+
+func (p Proof) Len() int { return len(p) }
+
+// equal is a temporary method that compares two proofs.
+// should be removed in BlobService V1.
+func (p Proof) equal(input Proof) error {
+	if p.Len() != input.Len() {
+		return ErrInvalidProof
+	}
+
+	for i, proof := range p {
+		pNodes := proof.Nodes()
+		inputNodes := input[i].Nodes()
+		for i, node := range pNodes {
+			if !bytes.Equal(node, inputNodes[i]) {
+				return ErrInvalidProof
+			}
+		}
+
+		if proof.Start() != input[i].Start() || proof.End() != input[i].End() {
+			return ErrInvalidProof
+		}
+
+		if !bytes.Equal(proof.LeafHash(), input[i].LeafHash()) {
+			return ErrInvalidProof
+		}
+
+	}
+	return nil
+}
+
+type jsonProof struct {
```

Comment:
are we using the Proof in rollkit? why not have Proof declared in proto along with Blob?

Reply:
Removed.